d3dx12.h fails MSVC analysis

Running MSVC static analysis with d3dx12.h results in the following warning:
`include\directx\d3dx12.h(2403): warning C6001: Using uninitialized memory '*pParameters_1_0.DescriptorTable.pDescriptorRanges'.`

To reproduce with the direct repro:
1. Generate the .sln with cmake and open it in VS
2. Open the properties for DirectX-Headers-Test, navigate to Code Analysis -> Microsoft and make sure the project is set to "Microsoft Native Minimum Rules"
3. Right click DirectX-Headers-Test and run "Analysis and Code Cleanup" -> "Run Code Analysis on DirectX-Headers-Test"

--- a/include/directx/d3dx12.h
+++ b/include/directx/d3dx12.h
@@ -2400,7 +2400,8 @@ inline HRESULT D3DX12SerializeVersionedRootSignature(
                         {
                             if (desc_1_1.pParameters[n].ParameterType == D3D12_ROOT_PARAMETER_TYPE_DESCRIPTOR_TABLE)
                             {
-                                HeapFree(GetProcessHeap(), 0, reinterpret_cast<void*>(const_cast<D3D12_DESCRIPTOR_RANGE*>(pParameters_1_0[n].DescriptorTable.pDescriptorRanges)));
+                                auto pDescriptorRanges_1_0 = pParameters_1_0[n].DescriptorTable.pDescriptorRanges;
+                                HeapFree(GetProcessHeap(), 0, reinterpret_cast<void*>(const_cast<D3D12_DESCRIPTOR_RANGE*>(pDescriptorRanges_1_0)));
                             }
                         }
                         HeapFree(GetProcessHeap(), 0, pParameters);

